deebot_client/commands/json/ota.py

```diff
@@ -0,0 +1,40 @@
+"""Ota command module."""
+from types import MappingProxyType
+from typing import Any
+
+from deebot_client.command import InitParam
+from deebot_client.event_bus import EventBus
+from deebot_client.events import OtaEvent
+from deebot_client.message import HandlingResult, MessageBodyDataDict
+
+from .common import JsonCommandWithMessageHandling, JsonSetCommand
+
+
+class GetOta(JsonCommandWithMessageHandling, MessageBodyDataDict):
+    """Get ota command."""
+
+    name = "getOta"
+    event_type = OtaEvent
+
+    @classmethod
+    def _handle_body_data_dict(
+        cls, event_bus: EventBus, data: dict[str, Any]
+    ) -> HandlingResult:
+        """Handle message->body->data and notify the correct event subscribers.
+
+        :return: A message response
+        """
+        event_bus.notify(cls.event_type(bool(data["autoSwitch"])))
```

Comment:
The deebot 950 Ozmo will not return the `autoSwitch` property but other data.
We should not expect that `autoSwitch` needs to be always present.

What is your bot returning on this command?

Reply:
Is it possible to turn off **auto ota** for **deebot 950 Ozmo**? 
I do not find the option for **950** in the APP and the result over API has also not that option (like you say **autoSwitch** not available).

The **deebot t10** has the option in the APP and over API. 
From what i see, this option is only available for newer bot's.

**t10** returns (https://github.com/DeebotUniverse/client.py/discussions/345#discussioncomment-7582167):
![image](https://github.com/DeebotUniverse/client.py/assets/7400017/cf497802-b6bb-43c0-9348-791b6251b94b)

**950** return:
![image](https://github.com/DeebotUniverse/client.py/assets/7400017/0e52154a-91d1-4342-abe3-7f1569aa4252)

Reply:
The 950 has no auto-update. You can only update it manually

Reply:
I think `supportAuto` indicates if auto-update is available or not.
We should adapt the `OtaEvent` to have at least the current version, the status and the progress.

Not sure what's the meaning of result and isForce

Reply:
What do you think about the updated version:

- extended with additional information for **get**
  - added: `support_auto`, `version`, `status`, `progress`
  - on inital **get** it is possible all the values could be filled
    - for older bots `support_auto` is set to `None` when not send and `enable` will by default `True`
    - for newer bots `support_auto` and `enable` will be converted as bool
- when **set** will be called to change `enable` state, only `autoSwitch` is respond by api
  - for that, all extended additional information, are set as None to identify no changes respond
  - for older bots, which can not change `autoSwitch`, they will have only **get** event and the state is always `enable=True`
- I'm not sure for older bots if they are auto updated or it is needed to manually trigger it in the app
  - Not saw updates in the app for long time
  - Here would be the question if `enable=True` is the correct state to show
- About the extended information `status` and `progress`
  - They will be only set by **get**
  - I'm not sure if there is an event when updates are happen